src/main/java/edu/harvard/iq/dataverse/AuxiliaryFileServiceBean.java

```diff
@@ -114,4 +119,32 @@ public AuxiliaryFile lookupAuxiliaryFile(DataFile dataFile, String formatTag, St
         }
     }
 
+    public List<AuxiliaryFile> findAuxiliaryFiles(DataFile dataFile) {
+        TypedQuery query = em.createQuery("select object(o) from AuxiliaryFile as o where o.dataFile.id = :dataFileId", AuxiliaryFile.class);
+        query.setParameter("dataFileId", dataFile.getId());
+        return query.getResultList();
+    }
+
+    public List<String> findAuxiliaryFileTypes(DataFile dataFile) {
+        Query query = em.createNativeQuery("select distinct type from auxiliaryfile where datafile_id = ?1");
+        query.setParameter(1, dataFile.getId());
+        return query.getResultList();
+    }
+
+    public List<AuxiliaryFile> findAuxiliaryFilesByType(DataFile dataFile, String typeString) {
+        TypedQuery query = em.createQuery("select object(o) from AuxiliaryFile as o where o.dataFile.id = :dataFileId and o.type like :type", AuxiliaryFile.class);
+        query.setParameter("dataFileId", dataFile.getId());
+        query.setParameter("type", typeString);
+        return query.getResultList();
+    }
+
```

Comment:
can we make these NamedQueries under AuxiliaryFile?

Reply:
Done in 3441758c9. I also moved existing queries.